### DISCUSSION TITLE
Redirection to Stripe Fails in SPA Mode

### DISCUSSION THREAD

**Discussion by Morgnus** [maartenpaauw/filament-cashier-billing-provider#22] [Q&A] [answered]
Hi everyone,

I'm currently working on a project using Cashier Billing Provider. I've encountered an issue where the redirection to Stripe for billing purposes doesn't work when SPA mode is enabled.

Has anyone faced a similar issue or found a workaround for this? Any insights or suggestions would be greatly appreciated.

Thank you in advance!

**Comment by maartenpaauw** [comment, marked as answer]
I didn't encounter this issue before. I've done applications using this package, but all aren't using SPA mode. I'll try it out with one of my applications later on. If I can't reproduce the bug, it feels more like a Filament PHP bug because that's where the redirection is handled.

**Comment by maartenpaauw** [comment]
I finally had the time to reproduce the bug locally with SPA mode. And indeed the redirect is not happening when SPA mode is enabled. Like I mentioned earlier, I rather think this is a Filament PHP bug because, the documentation tells use:

> `getSubscribedMiddleware()` returns the name of a middleware that should be used to check if the tenant has an active subscription. This middleware should redirect the user to the billing page if they don't have an active subscription.

And redirecting is what we do at the following line: https://github.com/maartenpaauw/filament-cashier-billing-provider/blob/main/src/Stripe/RedirectIfUserNotSubscribed.php#L60

Are you in the position to open up a GitHub issue at https://github.com/filamentphp/filament?
